dotnet/src/SemanticKernel/AI/OpenAI/Clients/OpenAIClientAbstract.cs

```diff
@@ -42,22 +48,26 @@ internal OpenAIClientAbstract(ILogger? log = null)
         this._httpClientHandler = new() { CheckCertificateRevocationList = true };
         this.HTTPClient = new HttpClient(this._httpClientHandler);
         this.HTTPClient.DefaultRequestHeaders.Add("User-Agent", HTTPUseragent);
+        // Setting the timeout to infinite.
+        // Will use cancellation token for timeout, giving users capability to set timeout per call.
+        this.HTTPClient.Timeout = Timeout.InfiniteTimeSpan;
```

Comment:
This makes me nervous, although I understand the intent. I worry about users adding new calls to HTTPClient in the future and not correctly passing a cancellation token. Would it make sense to modify the HTTPClient timeout when the call is about to be made with a valid cancellation token?